File: python/task1/S1_VAR_05.py
# S1_VAR_05 - Truthiness

def is_truthy(v):
    return bool(v)

test_values = [0, 1, "", "0", [], [0], {}, None]

print(f"{'Value':<12} {'Truthy?'}")
print("-" * 22)
for v in test_values:
    print(f"{repr(v):<12} {is_truthy(v)}")

File: js/task1/S1_VAR_05.js
// S1_VAR_05 - Array vs object
// typeof [] returns "object", so we need Array.isArray()

function isArray(value) {
  return Array.isArray(value);
}

// Test cases
const testValues = [
  { label: "[]", value: [] },
  { label: "[1, 2, 3]", value: [1, 2, 3] },
  { label: "{}", value: {} },
  { label: "{ a: 1 }", value: { a: 1 } },
  { label: "null", value: null },
  { label: "'string'", value: "string" },
  { label: "new Array()", value: new Array() }
];

console.log("Value".padEnd(15) + "typeof".padEnd(12) + "isArray()");
console.log("-".repeat(38));

for (const t of testValues) {
  const typeofResult = t.value === null ? "object" : typeof t.value;
  console.log(
    t.label.padEnd(15) +
    typeofResult.padEnd(12) +
    String(isArray(t.value))
  );
}
